crates/sui-types/src/transaction.rs

```diff
@@ -2054,6 +2036,22 @@ impl TransactionData {
     }
 }
 
+pub fn is_system_tx(kind: &TransactionKind, sender: SuiAddress) -> bool {
```

Comment:
Is the primary reason we have to make this a standalone function instead of a on a `self` due to the `OnChainTransactionInfo` data structure? If so, probably not worth  adapting to that because we are deprecating `OnChainTransactionInfo` soon.
Also, now that you could create a system transaction using PTB, it seems that the concept of system transaction no longer has anything to do with TransactionKind anymore. The is_system_tx() check now would literally become "sender == SuiAddress::ZERO". So if we are going this route, perhaps a cleaner way is to just add a function to `SuiAddress` called "is_system_address"? And then we unexpose ZERO and replace with a constant called "SYSTEM_ADDRESS" or some sort?

Reply:
Should I just special case `OnChainTransactionInfo` for now, until it is deleted?

And yes, it does seem we can remove the match and just test the sender address.

Reply:
It turns out the match is probably necessary (or else we need to do some bigger changes to the validation path), since we need to prevent users from sending (say) a ChangeEpoch tx with a non-zero sender address